src/context/WalletProvider/Ledger/components/Chains.tsx

```diff
@@ -52,7 +52,8 @@ export const LedgerChains = () => {
 
   const handleConnectClick = useCallback(
     async (chainId: ChainId) => {
-      if (!walletState?.wallet) {
+      const { wallet, deviceId } = walletState ?? {}
```

Comment:
i.e `state` should never be undefined in `IWalletContext`, as it is instantiated to an `InitialState` 

<img width="666" alt="image" src="https://github.com/user-attachments/assets/eeb7facf-a816-4699-b451-34d0b325bb1f" />

```suggestion
      const { wallet, deviceId } = walletState
```

Reply:
very good, here 1d9cd185c5f9c4dd9bd94e3ee40b63fbe846dd2e